pkg/controllers/dashboard/helm/repo.go

```diff
@@ -196,27 +202,40 @@ func (r *repoHandler) download(repoSpec *catalog.RepoSpec, status catalog.RepoSt
 	}
 
 	downloadTime := metav1.Now()
-	if repoSpec.GitRepo != "" && status.IndexConfigMapName == "" {
-		commit, err = git.Head(secret, metadata.Namespace, metadata.Name, repoSpec.GitRepo, repoSpec.GitBranch, repoSpec.InsecureSkipTLSverify)
-		if err != nil {
-			return status, err
+	if repoSpec.GitRepo != "" {
+		// the helm repo is hosted in a git repo
+		if status.IndexConfigMapName == "" {
+			// it is the first time to download this repo
+			commit, err = git.Head(secret, metadata.Namespace, metadata.Name, repoSpec.GitRepo, repoSpec.GitBranch, repoSpec.InsecureSkipTLSverify)
+			if err != nil {
+				return status, err
+			}
+		} else {
+			commit, err = git.Update(secret, metadata.Namespace, metadata.Name, repoSpec.GitRepo, repoSpec.GitBranch, repoSpec.InsecureSkipTLSverify)
+			if err != nil {
+				return status, err
+			}
+			// check if the IndexConfigMap exists
+			exist := true
+			_, err = r.configMaps.Get(status.IndexConfigMapNamespace, status.IndexConfigMapName, metav1.GetOptions{})
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					exist = false
+				} else {
+					return status, err
+				}
+			}
+			if status.Commit == commit && exist {
+				status.DownloadTime = downloadTime
+				return status, nil
+			}
```

Comment:
nit: the `exists` variable is not needed.
```suggestion
			if err != nil {
				if !apierrors.IsNotFound(err) {
					return status, err
				}
			} else if status.Commit == commit {
				status.DownloadTime = downloadTime
				return status, nil
			}
```

Reply:
hmm, your suggestion will miss this case:
 update the downloadTime and return status if configMap exists and "status.Commit == commit"

Reply:
updated.